plugins/woocommerce/client/blocks/assets/js/blocks/product-description/edit.tsx

```diff
@@ -0,0 +1,219 @@
+				kind,
+				name,
+				id: recordId,
+			} ),
+		[ kind, name, recordId ]
+	);
+}
+
+function ReadOnlyContent( {
+	parentLayout,
+	layoutClassNames,
+	userCanEdit,
+	postType,
+	postId,
+} ) {
+	const [ , , content ] = useEntityProp(
+		'postType',
+		postType,
+		'content',
+		postId
+	);
+	const blockProps = useBlockProps( { className: layoutClassNames } );
+	const blocks = useMemo( () => {
+		return content?.raw ? parse( content.raw ) : [];
+	}, [ content?.raw ] );
+	const blockPreviewProps = useBlockPreview( {
+		blocks,
+		props: blockProps,
+		layout: parentLayout,
+	} );
+
+	if ( userCanEdit ) {
+		/*
+		 * Rendering the block preview using the raw content blocks allows for
+		 * block support styles to be generated and applied by the editor.
+		 *
+		 * The preview using the raw blocks can only be presented to users with
+		 * edit permissions for the post to prevent potential exposure of private
+		 * block content.
+		 */
+		return <div { ...blockPreviewProps }></div>;
+	}
+
+	return content?.protected ? (
+		<div { ...blockProps }>
+			<Warning>
+				{ __( 'This content is password protected.', 'woocommerce' ) }
+			</Warning>
+		</div>
+	) : (
+		<div
+			{ ...blockProps }
+			dangerouslySetInnerHTML={ { __html: content?.rendered } }
+		></div>
+	);
+}
+
+function EditableContent( { context = {} } ) {
+	const { postType, postId } = context;
+
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
+		'postType',
+		postType,
+		{ id: postId }
+	);
+
+	const entityRecord = useSelect(
+		( select ) => {
+			return select( coreStore ).getEntityRecord(
+				'postType',
+				postType,
+				postId
+			);
+		},
+		[ postType, postId ]
+	);
+
+	const hasInnerBlocks = !! entityRecord?.content?.raw || blocks?.length;
+
+	const initialInnerBlocks = [ [ 'core/paragraph' ] ];
+
+	const props = useInnerBlocksProps(
+		useBlockProps( { className: 'product-description__editable-content' } ),
+		{
+			value: blocks,
+			onInput,
+			onChange,
+			template: ! hasInnerBlocks ? initialInnerBlocks : undefined,
+		}
+	);
+	return <div { ...props } />;
+}
+
+function Content( props ) {
+	const { context: { queryId, postType, postId } = {}, layoutClassNames } =
+		props;
+	const blockProps = useBlockProps( { className: layoutClassNames } );
+
+	const userCanEdit = useCanEditEntity( 'postType', postType, postId );
+	if ( userCanEdit === undefined ) {
+		return null;
+	}
+
+	// Show warning if the block is not used in a product context.
+	if ( postType && postType !== 'product' ) {
+		return (
+			<div { ...blockProps }>
+				<Warning>
+					{ __(
+						'This block requires a product context to function. You can use it inside a Single Product template, Single Product block, Product Collection block, or within other product-related blocks.',
+						'woocommerce'
+					) }
+				</Warning>
+			</div>
+		);
+	}
+
+	const isDescendentOfProductCollection = Number.isFinite( queryId );
+	const isEditable = userCanEdit && ! isDescendentOfProductCollection;
```

Comment:
Any specific reason you decided to disable editing for its use within the product collection block, but keep it within the single product block?

Reply:
Okay, good question. TBH, I don't have any specific reason. It was there for Query Loop for Post Content block. So, I assumed that there was a reason behind not making it editable for Query Loop block. So, I thought I should not make it editable for product collection block as well. But, I don't see any good reason for this, TBH. So, I think we should remove this condition. So, we can allow users to edit description in the product collection block. 

Out of curiosity, I looked at the Gutenberg code and looks like Bart made this change in https://github.com/WordPress/gutenberg/pull/48642. CC: @WunderBart was there any specific reason for adding this condition? i.e.
```tsx
const isDescendentOfQueryLoop = Number.isFinite( queryId );
```

Reply:
I removed the condition in [d3dc6e9](https://github.com/woocommerce/woocommerce/pull/56202/commits/d3dc6e93df1bbf39086b7aab3c45093db3af5eb2)

Reply:
Makes me wonder if there is a performance issue in something like a query loop as it would have to pull in quite a few descriptions, and descriptions may not be overly small 🤷‍♂️ 
Having said that, a user would have to explicitly add it within the query loop, so I did be ok with removing the condition as you did.

Reply:
Even if we don't make it editable, the product collection will still fetch the product descriptions if they are included within the loop. So, there maybe a performance issue if it is used in the product collection. 

As you mentioned, users must explicitly include it in the Product collection block. For now, we can remove this condition, and should any performance issues occur later, we can reassess it. Does this sound good to you?